Add SPEC9 - Governance (#323)
scientific-python/specs: spec-0009/index.md

---
title: "SPEC 9 — Governance"
number: 9
date: 2024-06-04
author:

>   - "Sanket Verma <svsanketverma5@gmail.com>"

Reviewer: ```suggestion
  - "Sanket Verma <svsanketverma5@gmail.com>"
  - "Daniel McCloy <dan@mccloy.info>"
```

discussion: https://discuss.scientific-python.org/t/spec-9-governance/1229

> endorsed-by:

Reviewer 2: What would it mean for a project to endorse this SPEC?

---

> ## Description
> 

Reviewer: ```suggestion
## Description

This SPEC describes what governance is, and offers recommendations for choosing a
governance model for open-source projects.

```

Open-source project communities are amalgamations of software and the humans who
use and maintain it. We have various systems, like governance models, which
define certain roles for participating members in the project, and a code of
conduct, which protects and fosters a safe and collaborative environment for

> the community.

Author: I like this. Thanks, @drammock!

Having a robust system in place ensures the long-term sustainability of the
project.

The OS communities interact and contribute through various mediums, such as
GitHub, a chat platform, social media, blog posts, etc., and having guidelines
for contributions helps maintain a continuous stream of contributions.

This SPEC outlines the definition of governance, various open-source governance
models, the steps to choose the right governance model, and what needs to be
done post-adoption.

Please note that these are recommendations and not bylaws. If you have
suggestions/thoughts, we're more than happy to consider them.

### Core Project Endorsement

<!--
Briefly discuss what it means for a core project to endorse this SPEC.
-->

### Ecosystem Adoption

> <!--

Reviewer 3: ```suggestion

```

> Briefly discuss what it means for a project to adopt this SPEC.

Reviewer 3: ```suggestion
The endorsement of this SPEC signifies your project's support for the 
guidelines laid out in the document.
```

Author: Thanks for the suggestion here, @InessaPawson.
Wouldn't this suggestion fit better in the core project adoption section?

Also, I'd appreciate it if someone could emphasize the difference b/w core project and ecosystem adoption. 

Does the ecosystem cover projects which are not formally [core projects](https://scientific-python.org/specs/core-projects/) under the Scientific Python umbrella but are related to the Scientific Python ecosystem in general?

CC: @jarrodmillman @stefanv

Reviewer: > Also, I'd appreciate it if someone could emphasize the difference b/w core project and ecosystem adoption

cf the (brief) discussion in #271

Reviewer 3: > Wouldn't this suggestion fit better in the core project adoption section?

@MSanKeys963 Relying on my understanding of the SPEC process, I believe this is the best section for the sentence. This doesn’t mean that **it is indeed** the right place for it.:))
@stefanv @jarrodmillman @bsipocz What are you thoughts?

> -->

Reviewer 3: ```suggestion

```

## Implementation

### Defining governance

Governance are set of rules or customs that defines who gets to do what (and how
they are supposed to do it).

When you define governance for a project, you need to identify a few things:

> - What roles does maintainers, core-team, contributors and community members

Reviewer 3: ```suggestion
- What roles does maintainers, core team, contributors, and community members
```

    effectiveness
- **External Stakeholders**
  - Consider how partnerships with other projects, organizations, or companies
    might influence governance
- **Legal and financial considerations**
  - If the project needs to be incorporated as a non-profit, foundation, or
    under a company
  - Assess the sources of funding and how they influence governance

### What to do after the adoption?

Once a project has adopted a governance model, the several important steps could
be followed to ensure smooth implementation and operation.

- **Document the governance model**
  - Clearly document the governance model, including roles, responsibilities,
    decision-making processes, and procedures for conflict resolution.
  - https://communityrule.info/templates/ offers an excellent interface where
    you can create a `.md` for your governance model
- **Communicate to the Community**
  - Make a public announcement about the adoption of the new governance model
  - Provide an overview of how the governance model works, why it was chosen,
    and what it means for the community
- **Implement the governance structure**
  - Assign initial roles and responsibilities

>   - Form any necessary commiittees, councils, or boards as defined by the

Reviewer 4: ```suggestion
  - Form any necessary committees, councils, or boards as defined by the
```

there are two `m`s, two `t`s, two `e`s, but only one `i` :)

Author: Thanks, @ivanov!

  - "Daniel McCloy <dan@mccloy.info>"
  - "Matt Haberland <mhaberla@calpoly.edu>"
discussion: https://discuss.scientific-python.org/t/spec-9-governance/1229
endorsed-by:
---

## Description

This SPEC describes what governance is, and offers recommendations for choosing a
governance model for open-source projects.

Open-source project communities are amalgamations of software and the humans who
use and maintain it. Governance models define methods for decision-making
within a project, roles for the community members imbued with decision-making
authority, and processes for filling (and, if needed, vacating) those
leadership roles. In the context of open-source software projects, governance
models often also define the granting of necessary permissions or access to
organization secrets necessary for the implementation of project-related
decisions. Governance models are complementary to Codes of Conduct and
Community Guidelines, which define the expectations and responsibilities of all
participants in a project (including users of the software), regardless of
their status as a project leader (cf. SPEC XXX), ensuring the long-term
sustainability of a project.

> The OS communities interact and contribute through various mediums, such as

Reviewer 5: ```suggestion
The open source communities interact and contribute through various mediums, such as
```
It looks like this is the only place "open source" (or the acronym) is needed; otherwise, I would suggest defining the acronym here.

There are few **pre-defined models** for governance used across open-source
projects. We'll mention them here for reference:

- BDFL
  - BDFL stands for 'Benevolent Dictator for Life.' Under this structure, one
    person (usually the project's initial author) has the final say on all
    major project decisions. Smaller projects are BDFL by default because
    there are only one or two maintainers.
  - BDFL model template → http://oss-watch.ac.uk/resources/benevolentdictatorgovernancemodel
- Meritocracy / Do-ocracy
  - Decisions are made by contributors who have demonstrated merit through
    their contributions to the project. Merit is often gained through
    code contributions, documentation improvements, or other significant efforts.
  - Meritocracy model template → http://oss-watch.ac.uk/resources/meritocraticgovernancemodel
- Self appointing council / committee / board
  - This model appoints a board or committee to govern various aspects of a
    project. Those groups are often called the steering committee, committer
    council, technical operation committee, board of directors, etc. This
    model might be helpful for projects that need a sponsoring foundation,

>     and establishing electoral mechanisms is challenging.

Reviewer 5: ```suggestion
    and projects for which establishing electoral mechanisms is challenging.
```

---
title: "SPEC 9 — Governance"
number: 9
date: 2024-06-04
author:
  - "Sanket Verma <svsanketverma5@gmail.com>"
  - "Inessa Pawson <inessapawson@gmail.com>"
  - "Daniel McCloy <dan@mccloy.info>"
  - "Matt Haberland <mhaberla@calpoly.edu>"

> discussion: https://discuss.scientific-python.org/t/spec-9-governance/1229

Reviewer 6: I would say add is-draft now, and then merge this PR for this last round of asking for inputs rather then keeping it open for an undefined time.

```suggestion
discussion: https://discuss.scientific-python.org/t/spec-9-governance/1229
is-draft: true
```

cc @stefanv

Author: +1 for merging this as a draft.

    their contributions to the project. Merit is often gained through
    contributions of code, documentation, or other significant efforts.
  - Meritocracy model template → http://oss-watch.ac.uk/resources/meritocraticgovernancemodel
- Self appointing council / committee / board
  - This model appoints a board or committee to govern various aspects of a
    project. Those groups are often called the steering committee, committer
    council, technical operation committee, board of directors, etc. This
    model might be helpful for projects that need a sponsoring foundation,
    and establishing electoral mechanisms is challenging.
- Electoral
  - The electoral governance model in open-source projects is a structured
    approach in which the community elects the leadership or decision-making
    body. This model aims to ensure that those who guide the project's
    direction and make crucial decisions are chosen democratically,
    representing the broader community's interests.
- Foundation backed
  - A non-profit foundation oversees the project, and its governance
    structures often include a board of directors, technical committees, and

>     working groups.

Author: I've removed the 'Corporate backed / Single Vendor governance model' in https://github.com/scientific-python/specs/pull/323/commits/750a088acb40e843817d94f770259618ae6f8222 as discussed with @jarrodmillman over the summit.

Should we also remove the 'Foundation-backed' model? I went through [Jupyter's old governance model](https://jupyter.org/governance/archive/governance.html) and realised there used to be a subcommittee for non-profit (NumFOCUS) to manage the funding.

I understand that it's the old model and not the current one, but I would like to know if it's helpful to keep it here. Are there any scientific Python projects out there that subscribe to a similar model?
